students/shibin_mathew/lesson6/test_mailroom.py

```diff
@@ -0,0 +1,67 @@
+import unittest
+from collections import defaultdict
+from mailroom import *
+
+
+
+class MailroomTestCase(unittest.TestCase):
+    # Tests for 'mailroom.py'
+    def test_switch_menu(self):
+        self.assertEqual(switch_menu('1'), send_thank_you)
+        self.assertEqual(switch_menu('2'), create_report)
+        self.assertEqual(switch_menu('3'), send_letters)
+        self.assertEqual(switch_menu('4'), quit_menu)
+
+    def test_send_letters(self):
+        list_test = []
+        list_test2 = []
+        #  Test 1
+        for names in donors2:
+            list_test.append(names+'.txt')
+        self.assertEqual(send_letters(donors2), list_test)
+        #  Test 2
+        for names in donors:
+            list_test2.append(names+'.txt')
+        self.assertEqual(send_letters(donors), list_test2)
+
+    def test_create_report(self):
+        list_test = []
+        list_test2 = []
+        #  Test 1
+        for names in donors2:
+            list_test.append([names, round(sum(donors2[names]), 2), len(donors2[names]),
+                             round(sum(donors2[names]) / len(donors2[names]), 2)])
+        self.assertEqual(create_report(donors2), list_test)
+        #  Test 2
+        for names in donors:
+            list_test2.append([names, round(sum(donors[names]), 2), len(donors[names]),
+                             round(sum(donors[names]) / len(donors[names]), 2)])
+        self.assertEqual(create_report(donors), list_test2)
```

Comment:
You're on the right track here, but see my note in mailroom about writing testable functions and avoiding return values that are only there for your tests